### PR TITLE
DNS switch to new net_context API

### PR DESCRIPTION
**Pull request by tbursztyka**

### CODE REVIEW

**Comment by tbursztyka** [comment]
mdns will follow shortly, so no hurry to apply that PR now.

File: include/net/net_context.h [hunk 1]
```diff
@@ -293,6 +293,15 @@ struct net_context {
 
 	/** Network interface assigned to this context */
 	s8_t iface;
+
+
```
Comment (jukkar) [inline include/net/net_context.h]: extra empty line

File: include/net/net_context.h [hunk 2]
```diff
@@ -293,6 +293,15 @@ struct net_context {
...
+	/** IPv6 hop limit or IPv4 ttl for this network packet.
```
Comment (jukkar) [inline include/net/net_context.h]: Do we really need to store this in net_context? The hop limit/ttl is already stored in net_if and we have interface index in net_context so we could fetch it from net_if directly.

**Review by jukkar — CHANGES_REQUESTED** [review]

File: include/net/net_context.h (hunk 2 above)
Comment (tbursztyka) [reply include/net/net_context.h]: What about temporary ttl/hop_limit tweak? As far as I can see, that's what is happening in resolve.c

**Comment by tbursztyka** [comment]
Note that I still need to test llmnr and mdns responder

File: include/net/net_context.h (hunk 2 above)
Comment (jukkar) [reply include/net/net_context.h]: Yep, you are right here, I missed the temp hop-limit setting in LLMNR

**Review by pfalcon — DISMISSED** [review]
Minor issues spotted so far, still reviewing.

File: include/net/net_context.h (hunk 2 above)
Comment (tbursztyka) [reply include/net/net_context.h]: Indeed, quickly  c/p from net_pkt

**Review by jukkar — APPROVED** [review]
LGTM

**Review by pfalcon — APPROVED** [review]
Based on the discussion in https://github.com/zephyrproject-rtos/zephyr/pull/13506, there's little choice but just approve it.

**Pull request merged**